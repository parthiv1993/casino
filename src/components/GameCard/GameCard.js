import React from 'react';
import PropTypes from 'prop-types';
import getLabel from '../../helpers/labels';
import Button from '../Buttton';
import Seperator from '../Seperator';
import Icon from '../Icon';
import StyledText from '../StyledText';
import {
  GameActionsContainer,
  GameCardContainer,
  GameDescription,
  GameThumbnail,
  GameTitle
} from './GameCard.styles';
import { baseURL } from '../../api/urls';
const GameCard = ({ name, code, description, icon, onPlay }) => {
  const thumbnailLink = `${baseURL}/${icon}`;

  return (
    <>
      <GameCardContainer>
        <GameTitle>{name}</GameTitle>
        <GameThumbnail src={thumbnailLink} alt={`${name}-thumbnail`}></GameThumbnail>
        <GameDescription>{description}</GameDescription>
        <GameActionsContainer>
          <Button
            buttonType={'secondary'}
            onClick={() => onPlay(code)}
            data-testid={`${code}-play`}>
            <StyledText type="subTitle">{getLabel('HOME_SCREEN_GAMES_PLAY')}</StyledText>
            <Icon iconName="fa-solid fa-chevron-right" addMargin />
          </Button>
        </GameActionsContainer>
      </GameCardContainer>
      <Seperator />
    </>
  );
};

GameCard.propTypes = {
  name: PropTypes.string,
  description: PropTypes.string,
  icon: PropTypes.string,
  onPlay: PropTypes.func,
  code: PropTypes.string
};

GameCard.defaultProps = {
  name: null,
  description: null,
  icon: null,
  code: null,
  onPlay: () => {}
};

export default React.memo(GameCard);
