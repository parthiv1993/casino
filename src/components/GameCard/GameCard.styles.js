import styled from 'styled-components';
import { devicesMediaQueries } from '../../helpers/deviceHelper';
import StyledText from '../StyledText';

export const GameCardContainer = styled.article`
  display: grid;
  grid-template-columns: fit-content(30%);
  grid-template-rows: auto;
  gap: 1rem;

  ${devicesMediaQueries.whenMobile} {
    display: block;
  }
`;

export const GameThumbnail = styled.img.attrs({ height: 'auto', width: '100%' })`
  grid-column: 1;
  grid-row: 1/3;
`;

export const GameTitle = styled(StyledText).attrs({
  type: 'title'
})`
  grid-column: 2;
  grid-row: 1;
  display: none;
`;

export const GameDescription = styled(StyledText).attrs({
  type: 'description'
})`
  grid-column: 2;
  grid-row: 2;
`;

export const GameActionsContainer = styled.div`
  grid-column: 2;
  grid-row: 3;
  text-align: right;
`;
