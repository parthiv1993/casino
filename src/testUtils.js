import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import light from './contexts/theme/light';

export const renderComponentWithLightTheme = (children) => {
  return render(<ThemeProvider theme={light}>{children}</ThemeProvider>);
};

export const lowerCaseAllKeys = (props = {}) => {
  const lowercaseObject = {};
  Object.keys(props).map((key) => (lowercaseObject[key.toLowerCase()] = props[key]));

  return lowercaseObject;
};

export const MockIcon = (props) => <div component="Icon" {...lowerCaseAllKeys(props)} />;
