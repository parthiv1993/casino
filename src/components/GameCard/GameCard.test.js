import { fireEvent } from '@testing-library/react';
import { renderComponentWithLightTheme } from '../../testUtils';
import GameCard from './GameCard';

const mockOnPlay = jest.fn();

const dummyGame = {
  name: 'gameName',
  code: 'gameCode',
  description: 'gameDescription',
  icon: 'gameIcon',
  onPlay: mockOnPlay
};

describe('GameCard', () => {
  it('should match snapshot', () => {
    const { container } = renderComponentWithLightTheme(<GameCard {...dummyGame} />);
    expect(container).toMatchSnapshot();
  });

  it('should call onPlay when clicked on Play button', () => {
    const { getByTestId } = renderComponentWithLightTheme(<GameCard {...dummyGame} />);
    const playButton = getByTestId('gameCode-play');
    fireEvent.click(playButton);
    expect(mockOnPlay).toHaveBeenCalledTimes(1);
    expect(mockOnPlay).toHaveBeenCalledWith('gameCode');
  });
});
