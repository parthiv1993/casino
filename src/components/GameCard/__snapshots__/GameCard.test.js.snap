// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameCard should match snapshot 1`] = `
.c7 {
  background: #1b1c1d;
  border-width: 0px;
  border-color: none;
  color: white;
  cursor: pointer;
  padding: 0.5em;
  border-radius: 4px;
}

.c7.c7:focus {
  outline: 2px solid blue;
}

.c10 {
  width: 100%;
}

.c9 {
  margin-inline: 4px;
}

.c1 {
  font-weight: bold;
  font-size: 1.5rem;
  margin: 0;
  display: inline-block;
}

.c8 {
  font-weight: 600;
  font-size: 1.25rem;
  margin: 0;
  display: inline-block;
}

.c4 {
  font-weight: normal;
  font-size: 1rem;
  margin: 0;
  display: inline-block;
}

.c0 {
  display: grid;
  grid-template-columns: fit-content(30%);
  grid-template-rows: auto;
  gap: 1rem;
}

.c3 {
  grid-column: 1;
  grid-row: 1/3;
}

.c2 {
  grid-column: 2;
  grid-row: 1;
  display: none;
}

.c5 {
  grid-column: 2;
  grid-row: 2;
}

.c6 {
  grid-column: 2;
  grid-row: 3;
  text-align: right;
}

@media (max-width:425px) {
  .c0 {
    display: block;
  }
}

<div>
  <article
    class="c0"
  >
    <h2
      class="c1 c2"
    >
      gameName
    </h2>
    <img
      alt="gameName-thumbnail"
      class="c3"
      height="auto"
      src="http://localhost:3001/gameIcon"
      width="100%"
    />
    <p
      class="c4 c5"
    >
      gameDescription
    </p>
    <div
      class="c6"
    >
      <button
        class="c7"
        data-testid="gameCode-play"
      >
        <h4
          class="c8"
        >
          Play
        </h4>
        <i
          class="c9 fa-solid fa-chevron-right"
        />
      </button>
    </div>
  </article>
  <hr
    class="c10"
  />
</div>
`;
